lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -615,14 +615,19 @@ fn generate_constraint_init_group(
                     }
                     let pa: #ty_decl = #from_account_info_unchecked;
                     if #if_needed {
+                        let mint_program_id = {
+                            let mint_account = #mint.to_account_info();
```

Comment:
`to_account_info()` internally uses `clone()` to get the `AccountInfo`. We can avoid cloning by using:
```suggestion
                            let mint_account: &AccountInfo = #mint.as_ref();
```